File: src/main/java/upbrella/be/rent/entity/ImprovementReport.java
package upbrella.be.rent.entity;

import lombok.*;

import javax.persistence.*;

@Entity
@Getter
@Builder
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
public class ImprovementReport {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    @OneToOne
    @JoinColumn(name = "history_id")
    private History history;
    private String content;
    private String etc;

    public static ImprovementReport createFromReturn(History history, String content) {

        return ImprovementReport.builder()
                .history(history)
                .content(content)
                .build();
    }
}

File: src/main/kotlin/upbrella/be/rent/entity/ImprovementReport.kt
package upbrella.be.rent.entity

import javax.persistence.*

@Entity
class ImprovementReport(
    @OneToOne
    @JoinColumn(name = "history_id")
    val history: History,
    val content: String? = null,
    val etc: String? = null,
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null
)
